### DISCUSSION TITLE
Writes for Portals

### DISCUSSION THREAD

**Discussion by jackfischer** [planetscale/discussion#256] [Feature Requests] [answered]
Currently, it seems like your application has to be aware of the write region versus the nearest read region for every query. This complicates application development and is a huge blocker for legacy applications. If portals could 1) automatically route write queries to primary in the background and 2) send subsequent reads to primary until we're sure everything is replicated.

Aware that (2) in particular is brushing over a lot of complexity. However if it is possible to make some assumptions and reduce this complexity, the feature would be enormously valuable.

**Comment by orware** [comment, marked as answer]
Thanks @jackfischer!

This is certainly something that could now be possible due to the new infrastructure introduced last week as part of our [serverless driver for JavaScript](https://planetscale.com/blog/introducing-the-planetscale-serverless-driver-for-javascript) release.

The following article from fly.io does a great job of describing the current implementation requirements on the application side of things to support multi-region using the current iteration of PlanetScale Portals:
https://fly.io/laravel-bytes/multi-region-laravel-with-planetscale/

But as you noted it does add some extra complexity which may not be possible in all cases to implement at the application layer.

I think the likelihood of this being simplified in the future is pretty good for users, but there's no official release information on when something like that would be available for users at this time.

I hope this helps!
